lib/roast/workflow/metadata_manager.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module Roast
+  module Workflow
+    # Manages workflow metadata, providing a structure parallel to output
+    # but specifically for internal metadata that shouldn't be user-facing
+    class MetadataManager
+      def initialize
+        @metadata = ActiveSupport::HashWithIndifferentAccess.new
+        @metadata_wrapper = nil
+      end
+
+      # Get metadata wrapped in DotAccessHash for dot notation access
+      def metadata
+        @metadata_wrapper ||= DotAccessHash.new(@metadata)
```

Comment:
DotAccessHash is really convenient but it adds another to access properties and I don't know if we need it for consistency. For instance, this [PR](https://github.com/Shopify/roast/blob/2bca1cb73696a64295dc4a6a6ac32d61a657e854/lib/roast/workflow/agent_step.rb#L27) does not use it and a IMO, few people would instinctively use the dot notation. Don't get me wrong, I like the dot notation but for consistency, its adding a new "standard"   in the codebase. What do you think ?